packages/next/client/dev-build-watcher.js

```diff
@@ -0,0 +1,205 @@
+import { getEventSourceWrapper } from './dev-error-overlay/eventsource'
+
+export default function initializeBuildWatcher () {
+  const shadowHost = document.getElementById('__next-build-watcher')
+  if (!shadowHost) return
+  let shadowRoot
+  let prefix = ''
+
+  if (shadowHost.attachShadow) {
+    shadowRoot = shadowHost.attachShadow({ mode: 'open' })
+  } else {
+    // If attachShadow is undefined then the browser does not support
+    // the Shadow DOM, we need to prefix all the names so there
+    // will be no conflicts
+    shadowRoot = shadowHost
+    prefix = '__next-build-watcher-'
```

Comment:
We can have the prefix included in browsers that support the Shadow DOM as well so there is no need to add `${prefix}` everywhere in the template strings, but then we would have to replace it with `__next-build-watcher-` to support Edge/IE. Not sure which approach is the most readable.